spring-vault-core/src/main/java/org/springframework/vault/core/lease/SecretLeaseContainer.java

```diff
@@ -463,10 +474,15 @@ protected Lease doRenewLease(RequestedSecret requestedSecret, final Lease lease)
 						@Override
 						@SuppressWarnings("unchecked")
 						public ResponseEntity<Map<String, Object>> doWithRestOperations(
-								RestOperations restOperations) {
+										RestOperations restOperations) {
```

Comment:
It should be sufficient to return just `null` for a `Lease` without `leaseId` and remove your additions. Everything else should happen itself.